src/chainstate/stacks/db/blocks.rs

```diff
@@ -5280,21 +5280,24 @@ impl StacksChainState {
         // epoch defined by this miner.
         clarity_tx.reset_cost(ExecutionCost::zero());
 
-        debug!("Evaluating block with epoch = {}", evaluated_epoch);
-        if evaluated_epoch >= StacksEpochId::Epoch21 {
+        // is this stacks block the first of a new epoch?
```

Comment:
Just curious why we're reversing the order of processing an epoch transition and the call to `Self::check_and_handle_reward_start`?

Reply:
I did this because I wanted the evaluated epoch before running check_and_handle_reward_start